conmon/conmon.c

```diff
@@ -582,18 +694,28 @@ int main(int argc, char *argv[])
 					pipe = STDOUT_PIPE;
 				else if (masterfd == masterfd_stderr)
 					pipe = STDERR_PIPE;
+				else if (oom_handling_enabled && masterfd == efd) {
+					if (read(efd, &oom_event, sizeof(uint64_t)) != sizeof(uint64_t))
+						nwarn("read eventfd");
+					ninfo("OOM received");
+					if (open("oom", O_CREAT, 0666) < 0) {
+						nwarn("failed to write oom file");
+					}
+				}
 				else {
 					nwarn("unknown pipe fd");
 					goto out;
 				}
 
-				num_read = read(masterfd, buf, BUF_SIZE);
-				if (num_read <= 0)
-					goto out;
+				if (masterfd == masterfd_stdout || masterfd == masterfd_stderr) {
+					num_read = read(masterfd, buf, BUF_SIZE);
+					if (num_read <= 0)
+						goto out;
```

Comment:
I know they're pre-existing, but can you use a break here instead of a goto?  The goto in write_k8s_log
() is ok in my book as you want to decrement the counter and try the next in the loop, but I don't think it's needed here and goto's have always made me twitch.  Too many beatings in school I guess.